### ISSUE TITLE
[A11y] User's bold text preference not respected if `TextSpan.style` has a custom font weight

### ISSUE THREAD

**Issue by loic-sharma** [flutter/flutter#177801]
iOS and Android have a system setting to bold text. This can be used to make text easier to read for accessibility purposes.

Flutter exposes this setting as [`PlatformDispatcher.accessibilityFeatures.boldText`](https://api.flutter.dev/flutter/dart-ui/AccessibilityFeatures/boldText.html), [`MediaQueryData.boldText`](https://api.flutter.dev/flutter/widgets/MediaQueryData/boldText.html), and [`MediaQuery.boldTextOf(context)`](https://api.flutter.dev/flutter/widgets/MediaQuery/boldTextOf.html). If this setting is on, `Text` and `Text.rich` should be bolded automatically, even if `Text`'s or `Text.rich`'s `style` specifies a custom font weight.

### Steps to reproduce

1. Run the code sample below on any platform (the app modifies `MediaQueryData` to mimic a system where the user has turned on bolded text)
1. The text should be bolded. However, it is not.

### Expected results

The text `Hello world` should be bolded. The user's preference for bold text should override the `TextSpan`'s font weight.

### Actual results

The text `Hello world` is not bolded.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: Scaffold(
        body: Builder(
          builder: (context) {
            // Modify MediaQueryData to mimic a system where the user
            // has turned on bolded text.
            return MediaQuery(
              data: MediaQuery.of(context).copyWith(boldText: true),  
              child: Builder(
                builder: (context) {
                  return const Center(
                    child: Text.rich(
                      TextSpan(
                        text: 'Hello world',
                        style: TextStyle(fontWeight: FontWeight.normal),
                      ),
                    ),
                  );
                }
              ),
            );
          }
        ),
      ),
    );
  }
}
```

</details>

**Comment by loic-sharma** [comment]
cc @Renzo-Olivares @LongCatIsLooong
